odoo/addons/base/models/ir_async.py

```diff
@@ -0,0 +1,265 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import datetime
+import itertools
+import json
+import logging
+import re
+import traceback
+import threading
+import odoo
+import odoo.exceptions
+from odoo import api, fields, models
+from odoo.http import serialize_exception
+
+_logger = logging.getLogger(__name__)
+BASE_VERSION = odoo.modules.load_information_from_description_file('base')['version']
+
+
+class IrAsync(models.Model):
+    _name = 'ir.async'
+    _description = 'Asynchronous Jobs'
+
+    name = fields.Char()
+    state = fields.Selection([
+        ('created', 'Created'),     # The job has been enqueued for later processing
+        ('succeeded', 'Succeeded'), # The job finished with a result, the result has not yet been retrieved
+        ('failed', 'Failed'),       # The job failed due to an exception
+        ('done', 'Done'),           # The job finished without result or the result has been retrieved
+    ])
+    res_model = fields.Char()
+    res_ids = fields.Text()
+    method = fields.Char()
+    args = fields.Text()
+    kwargs = fields.Text()
+    user_id = fields.Many2one('res.users', string='Scheduler User', ondelete='cascade')
+    context = fields.Text()
+    super_user = fields.Boolean()
+    traceback = fields.Text()
+    payload = fields.Text()
+
+    def call(self, method, *args, **kwargs):
```

Comment:
why public?

Reply:
made private

Reply:
Seems to still be public. So's `call_notify`.

Also not fond of the naming, `call` doesn't really imply that we're enqueing a job which will complete at some point in the future.